index.js

```diff
@@ -373,7 +375,24 @@ window.randomShape = () => {
   let layers = getRandomInt(maxLayer);
   let code = '';
   for (var i = 0; i <= layers; i++) {
-    code = code + getRandomShape() + getRandomColor() + getRandomShape() + getRandomColor() + getRandomShape() + getRandomColor() + getRandomShape() + getRandomColor() + ':';
+	let layertext = '';
+	for (var y = 0; y <= 3; y++) {
+		let randomShape = getRandomShape();
+		let randomColor = getRandomColor();
+		
+		if(randomShape === '-') {
+			randomColor = '-';
+			console.log('in');
+		}
+		layertext = layertext + randomShape + randomColor;
+	}
+	//empty layer not allowed
+	if(layertext === '--------'){
+		layertext = '';
+		y--;
```

Comment:
You made a small error here; in 392, y is no longer in scope, and 391 is unnecessary given that you're declaring the variable at the start of the loop. try replacing it with `i--`

Reply:
👍 thx